File: src/main/java/name/gyger/jmoney/account/AccountController.java
package name.gyger.jmoney.account;

import org.springframework.web.bind.annotation.*;

import java.util.Collection;

@RestController
@RequestMapping("/rest/accounts")
public class AccountController {

    private final AccountService accountService;

    public AccountController(AccountService accountService) {
        this.accountService = accountService;
    }

    @RequestMapping(method = RequestMethod.GET)
    public Collection<Account> getAccounts() {
        return accountService.getAccounts();
    }

    @RequestMapping(path = "/{accountId}", method = RequestMethod.GET)
    public Account getAccount(@PathVariable long accountId) {
        return accountService.getAccount(accountId);
    }

    @RequestMapping(path = "/{accountId}", method = RequestMethod.PUT)
    public void updateAccount(@RequestBody Account account, @PathVariable long accountId) {
        account.setId(accountId);
        accountService.updateAccount(account);
    }

    @RequestMapping(method = RequestMethod.POST)
    public long createAccount(@RequestBody Account account) {
        return accountService.createAccount(account);
    }

    @RequestMapping(path = "/{accountId}", method = RequestMethod.DELETE)
    public void deleteAccount(@PathVariable long accountId) {
        accountService.deleteAccount(accountId);
    }

}

File: src/main/kotlin/name/gyger/jmoney/account/AccountController.kt
package name.gyger.jmoney.account

import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/rest/accounts")
class AccountController(private val accountService: AccountService,
                        private val accountRepository: AccountRepository) {

    @GetMapping
    fun getAccounts(): Iterable<Account> {
        return accountRepository.findAll()
    }

    @GetMapping("/{accountId}")
    fun getAccount(@PathVariable accountId: Long): Account? {
        return accountRepository.findOne(accountId)
    }

    @PutMapping("/{accountId}")
    fun updateAccount(@RequestBody account: Account, @PathVariable accountId: Long) {
        account.id = accountId
        accountRepository.save(account)
    }

    @PostMapping
    fun createAccount(@RequestBody account: Account): Long {
        return accountService.createAccount(account)
    }

    @DeleteMapping("/{accountId}")
    fun deleteAccount(@PathVariable accountId: Long) {
        accountRepository.delete(accountId)
    }

}
